mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1002,19 +1019,20 @@ private[spark] object RandomForest extends Logging with Serializable {
       val numSplits = metadata.numSplits(featureIndex)
 
       // get count for each distinct value except zero value
-      val partNumSamples = featureSamples.size
-      val partValueCountMap = scala.collection.mutable.Map[Double, Int]()
-      featureSamples.foreach { x =>
-        partValueCountMap(x) = partValueCountMap.getOrElse(x, 0) + 1
-      }
+      val (partValueCountMap, partNumSamples) =
+        featureSamples.foldLeft((Map.empty[Double, Double], 0.0)) {
```

Comment:
I don't know how much it matters, but it might be less efficient to fold and update an immutable map here?

Reply:
done! updated to use mutable map